### PR TITLE
Fix: Two digit bug input

### LINKED ISSUE #83: Balance field - interupted when trying to enter two digits
**Description**
When I try to enter amount in balance field and delete "0.0" As soon as I enter one digit the ".0" is being added, In order to add 2 digits number I have to navigate cursor to the left 			
				
**Steps to Reproduce**
1. SpinUp Localhost env 
2. Open localhost:3000  
3. Check if Landing Page is loaded   
4. Click on REPuild box  
5. Click on Join button 
6. Enter two digit amount of tokens in balance field

**Expected behavior**
Should be able to enter two digits number

**Screenrecord**
https://user-images.githubusercontent.com/37701286/176949510-7ff3cc9d-6c80-4687-91a3-eef2a219a497.mp4

**Device Information**
OS: MacOs
Browser GoogleChrome, Brave
Device: MacOS Monterey v12.4

**Additional context**
Added screenrecord for better understanding.

### LINKED ISSUE #204: Amount field - unable to insert data in some cases
**Description**
Basically if you click anywhere besides center of the input field, the cursor will appear slightly off and you won't be able to enter data by typing it in. 

**Steps to Reproduce**
1. SpinUp Localhost env
2. Open localhost:3000
3. Check if Landing Page is loaded
4. Connect wallet
5. Click on SWPRGuild box
6. Click on 'Create Proposal'
7. Click on 'Vesting contract for DXD token' button
8. Click on 'Create vesting contract'
9. Click slightly of on 'Amount' field (check screenrecord)

**Expected behavior**
Should only have one place for for input in field 

**Screenrecord**
https://user-images.githubusercontent.com/37701286/184182336-00d84fd3-753f-444f-8923-bb48736866c8.mp4

**Device Information**
OS: MacOs
Browser: GoogleChrome, Brave
Device: MacOS Monterey v12.4

**Additional context**
This seems to be the case with every input field I've tried. (amount, recepient, function name etc.)

### PR DESCRIPTION
**Pull request by dcrescimbeni**
# Description

**Closes #83** 

Previous behaviour: when the `TokenAmountInput` field was empty and a user tried to input a number, after the first keystroke, the text was formatted to "X.0" and the caret jumped to the end of the string.

Now there's a 350ms delay between the last keystroke and the `value` assignment (and formatting). 

https://www.loom.com/share/9ee19481f0934329a2d0a96c58f0137c

**Closes #204** 

Clicking anywhere inside the `TokenAmountInput` component will either focus (click) or select its content (double click).

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Manual testing

# Checklist:

- [x] My code follows the existing style of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any UI changes have been tested and made responsive for mobile views

### CODE REVIEW

**Comment by Filipv95** [comment]
@dcrescimbeni @rossneilson Check this and it looks good! If you agreed that 350ms delay is enough for entering value that is fine by me also ^^

File: src/old-components/Guilds/common/Form/TokenAmountInput.tsx [hunk 1]
```diff
@@ -15,33 +16,35 @@ const TokenAmountInput: React.FC<TokenAmountInputProps> = ({
...
-    if (amountFromProps != null) {
-      setAmount(amountFromProps.toString());
+    if (localAmount !== valueAsString) {
+      setLocalAmount(valueAsString);
     }
-  }, [amountFromProps]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [valueAsString]);
 
   // Call onChange when the amount changes
```
Comment (Kenny-Gin1) [inline src/old-components/Guilds/common/Form/TokenAmountInput.tsx:32]: 🧹 Maybe remove this comment. I think it is self-explanatory through the code.

**Review by Kenny-Gin1 — COMMENTED** [review]
Looks good! I will approve this after @MiltonTulli huge PR. What were your conclusions about formatting the input fields after the user had unfocused on the input box instead? E.g. `onBlur`.

**Comment by dcrescimbeni** [comment]
> Looks good! I will approve this after @MiltonTulli huge PR. What were your conclusions about formatting the input fields after the user had unfocused on the input box instead? E.g. `onBlur`.

I decided to not do anything with it, since the delay to format the string is very short. My rationale was the following:
- I had to try to click on "submit" very fast to submit the form before the input was formatted.
- We have the case for just hitting "Enter", that submits the form without losing focus.

I think a normal user interacting with the form won't submit it before the format and value update kicks in.

File: src/old-components/Guilds/common/Form/TokenAmountInput.tsx (hunk 1 above)
Comment (dcrescimbeni) [reply src/old-components/Guilds/common/Form/TokenAmountInput.tsx]: Done!

**Review by Kenny-Gin1 — APPROVED** [review]

**Pull request merged**